### ISSUE TITLE
Exercise 2.2 – isSorted not as expected

### ISSUE THREAD

**Issue by jbspeakr** [fpinscala/fpinscala#538]
Exercise 2.2 requires to write a body for following function:

```
def isSorted[A](as: Array[A], ordered: (A,A) => Boolean): Boolean = ???
```

I expect this polymorphic function to return true if the sequence of elements in the provided array fulfil the provided `ordered` comparison. Unfortunately, the current "solution" does exactly the opposite...

```
  def isSorted[A](as: Array[A], gt: (A,A) => Boolean): Boolean = {
    @annotation.tailrec
    def go(n: Int): Boolean =
      if (n >= as.length-1) true
      else if (gt(as(n), as(n+1))) false
      else go(n+1)

    go(0)
  }
``` 

... will result in 

```
isSorted(Array(1,2,3), (x: Int, y: Int) => x <= y) # False
isSorted(Array(3,2,1), (x: Int, y: Int) => x <= y) # True
```

I'll open a PR.

**Comment by qqupp** [comment]
>  def isSorted[A](as: Array[A], gt: (A,A) => Boolean): Boolean = {

I believe that gt stands for `greater than` and so the specific instance for Int should be the something like this
```
 val gt = (x: Int, y: Int) => x > y
```
that results in 

```
scala> isSorted(Array(1,2,3), gt)
res0: Boolean = true
```

**Comment by brandshaide** [comment]
No, as this (`isSorted`) should be a example of a polymorphic function, the signature should not tell the caller to invoke a specific behaviour (like `greater than`). Me and @jbspeakr openend (almost) the exact same PR to take care of this issue. But it's not only a term of misleading/wrong signature of the function. The origin implementation is not behaving correctly - you can refer to both mine and @jbspeakr PR for further details.

**Comment by AndrewOdiit** [comment]
So does anyone have the correct implementation because not being able to do this is killing me !

**Comment by AndrewOdiit** [comment]
No worries , found a working version.
In case anyone is interested this worked for me:

 def isSorted[A](ar: Array[A], f:(A, A) => Boolean): Boolean = {
  
    @annotation.tailrec
    def loop(n: Int): Boolean =
      if (n == ar.length - 1) true
      else if (f(ar(n), ar(n + 1))) loop(n + 1)
      else false

    loop(0)
  }                                               //> isSorted: [A](ar: Array[A], f: (A, A) => Boolean)Boolean
  
  isSorted( Array(3,2,1) ,(x:Int, y:Int) => x <=y )

**Comment by brandshaide** [comment]
@AndrewOdiit me and @jbspeakr already submited a PR in order to fix this issue - but has not been merged yet.

**Comment by brandshaide** [comment]
@AndrewOdiit  https://github.com/fpinscala/fpinscala/pull/527
